src/fromager/sources.py

```diff
@@ -199,12 +200,16 @@ def download_url(
         return outfile
     # Open the URL first in case that fails, so we don't end up with an empty file.
     logger.debug(f"reading from {url}")
-    with requests.get(url, stream=True) as r:
-        r.raise_for_status()
-        with open(outfile, "wb") as f:
-            logger.debug(f"writing to {outfile}")
-            for chunk in r.iter_content(chunk_size=1024 * 1024):
-                f.write(chunk)
+
+    # Initialize the HTTPClient module
+    client = HTTPClient(url)
+    # Call HTTP GET for the URL
+    r = client.get_request()
```

Comment:
We lost the `stream=True` argument there, which is important for making the loop below work. Without that argument requests downloads the whole file in one go, which means you have to be able to hold the whole thing in RAM. We're downloading very large wheels in some case, so we don't want to allocate that much RAM at one time.

Reply:
Okay, I will make the changes and try to design this in a way @tiran suggested above